lib/mutant/runner/sink.rb

```diff
@@ -1,107 +1,72 @@
 module Mutant
   class Runner
-    # Abstract base class for computation sinks
     class Sink
-      include AbstractType
+      include Concord.new(:env)
 
-      # sink status
+      # Initialize object
       #
-      # @return [Object]
+      # @return [undefined]
       #
       # @api private
-      abstract_method :status
+      def initialize(*)
+        super
+        @start           = Time.now
+        @subject_results = {}
+      end
 
-      # Test if computation should be stopped
+      # Runner status
+      #
+      # @return [Result::Env]
+      #
+      # @api private
+      def status
+        Result::Env.new(
+          env:             env,
+          runtime:         Time.now - @start,
+          subject_results: @subject_results.values
+        )
+      end
+
+      # Test if scheduling stopped
       #
       # @return [Boolean]
       #
       # @api private
-      abstract_method :stop?
+      def stop?
+        env.config.fail_fast && !status.subject_results.all?(&:success?)
```

Comment:
Since `env` is reachable within the `status` object and `status.subject_results.all?(&:success?)` is operating on `status`'s dependencies could this be extracted to a simpler method in `Result::Env`?